### PR TITLE
Pass first argument of action parameters as open state of promises

### PR DESCRIPTION
**Pull request by nhunzaker**
Name says it all

```javascript
function getPlanet (id) {
  return fetch(`/planets/${id}`).then(res => res.json())
}

let action = repo.push(getPlanet, 'mars')

action.onOpen(data => console.log(data))
// 'mars'

action.onDone(data => console.log(data))
// { id: 'mars', name: 'Mars' }
```

Related issues:
https://github.com/vigetlabs/microcosm/issues/296

### CODE REVIEW

File: src/coroutine.js [hunk 1]
```diff
@@ -38,7 +38,9 @@ function processGenerator (action, body, repo) {
  * Coroutine is used by an action to determine how it should resolve
  * the body of their associated command.
  */
-export default function coroutine (action, body, repo) {
+export default function coroutine (action, command, params, repo) {
+  let body = command.apply(action, params)
```
Comment (efatsi) [inline src/coroutine.js]: noticed this went from `command.apply(null, params)` to `command.apply(action, params)`, why is that?

Comment (nhunzaker) [inline src/coroutine.js]: Very good catch. No reason. This would have been a bug.

Comment (nhunzaker) [inline src/coroutine.js]: Fixed with 766312e

**Comment by nhunzaker** [comment]
@greypants You can test this in `microcosm@12.7.0-alpha.2`. Could you give it a shot?

**Pull request merged**